client/components/domains/register-domain-step/index.jsx

```diff
@@ -553,27 +538,113 @@ const RegisterDomainStep = React.createClass( {
 		}
 
 		return mapDomainUrl;
-	},
+	}
 
-	goToMapDomainStep: function( event ) {
+	goToMapDomainStep = ( event ) => {
 		event.preventDefault();
 
-		this.recordEvent( 'mapDomainButtonClick', this.props.analyticsSection );
+		this.props.mapDomainButtonClick( this.props.analyticsSection );
 
 		page( this.getMapDomainUrl() );
-	},
+	};
 
-	showValidationErrorMessage: function( domain, error ) {
+	showValidationErrorMessage( domain, error ) {
 		const { message, severity } = getAvailabilityNotice( domain, error );
 		this.setState( { notice: message, noticeSeverity: severity } );
 	}
-} );
+}
 
-module.exports = connect( ( state, props ) => {
-	const queryObject = getQueryObject( props );
-	return {
-		currentUser: getCurrentUser( state ),
-		defaultSuggestions: getDomainsSuggestions( state, queryObject ),
-		defaultSuggestionsError: getDomainsSuggestionsError( state, queryObject ),
-	};
-} )( localize( RegisterDomainStep ) );
+const mapDomainButtonClick = ( section ) => composeAnalytics(
+	recordGoogleEvent(
+		'Domain Search',
+		'Clicked "Map it" Button'
+	),
+	recordTracksEvent(
+		'calypso_domain_search_results_mapping_button_click',
+		{ section }
+	),
+);
+
+const searchFormSubmit = ( searchBoxValue, section, count, vendor ) => composeAnalytics(
```

Comment:
I found the names of these analytics functions a bit misleading - I get it you used the analytics event to name them, but if I see a call to `this.props.searchFormSubmit`, I would assume it's actually for submitting some form? I think just prefixing them with `record`, like `recordSearchFormSubmit` will do the trick in most (all?) cases. Agree, disagree, `:partyparrot-middle-finger:`? :trollface: